app/src/main/java/at_challenge/deaguiar/daniel/atchalleng/controller/RouteDetailActivity.java

```
@@ -0,0 +1,203 @@
+
+                try {
+                    // Set BusStop Adapter
+                    mBusStopList = new BusStopList(resultBus);
+                    BusStopAdapter busAdapter = new BusStopAdapter(mBusStopList.getBusStopList());
+                    mStopListView.setAdapter(busAdapter);
+
+                    // List of Departures
+                    mDepartureList = new DepartureList(resultDepart);
+
+                    // set Weekday Departure Adapter
+                    ArrayList<Departure> weekdayDeparts = mDepartureList.getDepartures("WEEKDAY");
+                    if (weekdayDeparts.isEmpty()) {
+                        weekdayDeparts.add(new Departure(0, "WEEKDAY", "NO DEPARTURES"));
+                    }
+                    DepartureAdapter weekdayAdapter = new DepartureAdapter(weekdayDeparts);
+                    mWeekdayListView.setAdapter(weekdayAdapter);
+
+                    // set Saturday Departure Adapter
+                    ArrayList<Departure> saturdayDeparts = mDepartureList.getDepartures("SATURDAY");
+                    if (saturdayDeparts.isEmpty()) {
+                        saturdayDeparts.add(new Departure(0, "SATURDAY", "NO DEPARTURES"));
+                    }
+                    DepartureAdapter saturdayAdapter = new DepartureAdapter(saturdayDeparts);
+                    mStaturdayListView.setAdapter(saturdayAdapter);
+
+                    // set Sunday Departure Adapter
+                    ArrayList<Departure> sundayDeparts = mDepartureList.getDepartures("SUNDAY");
+                    if (sundayDeparts.isEmpty()) {
+                        sundayDeparts.add(new Departure(0, "SUNDAY", "NO DEPARTURES"));
+                    }
+                    DepartureAdapter sundayAdapter = new DepartureAdapter(sundayDeparts);
+                    mSundayListView.setAdapter(sundayAdapter);
+                } catch (JSONException e) {
+                    e.printStackTrace();
```

It's not good to swallow exceptions like this. Couldn't you handle that in the `BusStopList` class?